src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -472,6 +460,31 @@ internal void Configure()
         [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
         internal abstract void DetermineReflectionPropertyAccessors(MemberInfo memberInfo, bool useNonPublicAccessors);
 
+        private void EnsureRequiredPropertyIsDeserializable()
+        {
+            if (!CanDeserialize)
+            {
+                bool isCollectionProperty = (EffectiveConverter.ConverterStrategy & (ConverterStrategy.Enumerable | ConverterStrategy.Dictionary)) != 0;
+                bool isRequiredParameterRespected = AssociatedParameter?.IsRequiredParameter is true && Options.RespectRequiredConstructorParameters;
+
+                if (isCollectionProperty)
```

Comment:
Is this really only improving the error message or is it expanding failures for more types? It seems to me that if we have a collection property and `RespectRequiredConstructorParameters` is set to false this is failing when it wasn't before this change.